libbeat/beat/beat.go

```diff
@@ -248,27 +249,39 @@ func (b *Beat) Run() error {
 		logp.Critical("Running the beat returned an error: %v", err)
 	}
 
+	return err
+}
+
+// Stop calls the beater Stop action.
+// It can happen that this function is called more then once.
+func (b *Beat) Stop() {
+	logp.Info("Stopping Beat")
+	b.BT.Stop()
+
 	service.Cleanup()
 
 	logp.Info("Cleaning up %s before shutting down.", b.Name)
 
 	// Call beater cleanup function
-	err = b.BT.Cleanup(b)
+	err := b.BT.Cleanup(b)
 	if err != nil {
 		logp.Err("Cleanup returned an error: %v", err)
 	}
-	return err
 }
 
-// Stop calls the beater Stop action.
-// It can happen that this function is called more then once.
-func (beat *Beat) Stop() {
-	logp.Info("Stopping Beat")
-	beat.BT.Stop()
+var callback sync.Once
+var exit int
+
+func (b *Beat) Signal() {
+	b.error = fmt.Errorf("Beat stop through signal")
```

Comment:
@andrewkroh @urso this is called when the beat exists through CTRL-C. What should the exit code be in your opinion? 1 or 0?